app/summary/page.tsx

```diff
@@ -0,0 +1,143 @@
+"use client";
+
+import untypedSubjectSummaries from "@/app/generated/subject-summaries.json";
+import { SummaryRow } from "@/app/types";
+import { useState, useRef, useEffect } from "react";
+
+const SUMMARY_DISPLAY_COUNT = 100;
+const subjectSummaries = untypedSubjectSummaries as unknown as SummaryRow[];
+const tableHeaders = [
+    {
+        key: "enrollmentTerm",
+        label: "Term"
+    },
+    {
+        key: "subjectArea",
+        label: "Subject"
+    },
+    {
+        key: "catalogNumber",
+        label: "Course Number"
+    },
+    {
+        key: "courseTitle",
+        label: "Course Name"
+    },
+    {
+        key: "instructorName",
+        label: "Professor"
+    },
+    {
+        key: "gradeAverage",
+        label: "Average Grade"
+    },
+    {
+        key: "gradeSpread",
+        label: "Grade Spread"
+    },
+    {
+        key: "gradedTotal",
+        label: "Total Graded"
+    },
+    {
+        key: "enrollmentTotal",
+        label: "Total Enrolled"
+    }
+]
+
+const Summary = () => {
+    const [cutoff, setCutoff] = useState(SUMMARY_DISPLAY_COUNT);
+    const [sortBy, setSortBy] = useState<keyof SummaryRow>("gradeAverage");
+    const [sortReverse, setSortReverse] = useState(true);
+    const [displaySummaries, setDisplaySummaries] = useState(subjectSummaries.slice(0, cutoff));
+    const observerRef = useRef<HTMLDivElement>(null);
+
+    const onHeaderClick = (key: keyof SummaryRow) => {
```

Comment:
nit: the convention i use for event handler names is `onVerbNoun`, so for consistency this should be named `onClickHeader` (e.g. `onSelectSubjectArea`, `onSelectResult`)